### DISCUSSION TITLE
VS Code extension requests to sign in before connecting to a codespace

### DISCUSSION THREAD

**Discussion by El-Roque** [community/community#30196] [Codespaces]
Hi everyone, new user here.

I've been working with the web version of VS Code, and a couple of days ago I installed the desktop version for Mac. 
Everything run smoothly until yesterday. I rebooted my computer (I hadn't closed the program before), and when I started the program, I got the following message: "Please sign in to the GitHub Codespaces extension before connecting to a codespace".
It doesn't let me do anything. I tried uninstalling the extension, but I couldn't.

Any help you can provide will be much appreciated.
Thanks!

<img width="455" alt="Screen Shot 2022-08-20 at 10 07 12" src="https://user-images.githubusercontent.com/108275553/185747535-8adfa9be-9824-49ba-b0ca-e2f193b830e5.png">

**Comment by softwaredevelop** [comment]
I don't know your problem (and I am a Linux user), but the Sign in process:

- [Create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token)
- Sign in to GitHub

![1](https://user-images.githubusercontent.com/61334390/185784636-f61987de-858e-45c4-8bcb-33887cb6eec8.png)

- Use your Personal Access Token

![2](https://user-images.githubusercontent.com/61334390/185784700-2331a254-02d4-4652-9450-8bbd6c61447b.png)

What is your PAT expiration date?

**Reply by El-Roque** [reply]
Hi, thanks for the answer.
I cannot find a way to get a sign in window for Github within VS Code. I cannot do anything at all. 
I tried uninstalling and installing VS Code again, but it's the same.

I didn't have a PAT. I created one, but I don't know how to use it.
Sorry, but I'm super new to Github (got here doing the CS50 Harvard course) and I don't know what to do.

Thanks!

**Comment by softwaredevelop** [comment]
if added [Codespace](https://marketplace.visualstudio.com/items?itemName=GitHub.codespaces) extension to VS Code the Remote Explorer/Codespace icon
![1 1](https://user-images.githubusercontent.com/61334390/185809095-9e5dede0-3d88-41d8-8282-aad57929cd19.png)
sign in and allow and insert PAT

[if generated PAT make sure to copy and save the new PAT because you won't be able to see again](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-token), if you forget your PAT create new one

> [To develop in a codespace directly in VS Code, you must install and sign into the Codespaces extension with your GitHub credentials. The Codespaces extension requires VS Code October 2020 Release 1.51 or later.](https://docs.github.com/en/codespaces/developing-in-codespaces/using-github-codespaces-in-visual-studio-code#prerequisites)

**Comment by El-Roque** [comment]
Hi,
I had to completely uninstall the app and the user files, and install everything again, and now it's working (but I don't know for how long, as I couldn't figure out what was wrong).

Thanks a lot for your help!
